### ISSUE TITLE
LP1937096: checkbox remote ctrl-c prompt is confusing

### ISSUE THREAD

**Issue by beliaev-maksim** [canonical/checkbox#129]
This issue was migrated from https://bugs.launchpad.net/checkbox-ng/+bug/1937096

## Summary
| Status 	| Created on 	| Heat 	| Importance 	| Security related 	|
|--------	|------------	|------	|------------	|------------------	|
| Triaged	| 2021-07-21 15:02:36	| 10	| Medium	| False  	|

## Description
when I do ctrl-c when checkbox-remote is running and connected to a session, the options are confusing:

What do you want to interrupt?                                                                                        │
│                                                                                                                                          │
│               ( ) Cancel the interruption and resume the session (ESC)                                                                   │
│               ( ) Kill test in progress and move on to next                                                                              │
│               ( ) Disconnect the master (Same as CTRL+C)                                                                                 │
│               ( ) Stop the checkbox slave @10.245.130.30                                                                                 │
│               (X) Abandon the session on the slave @10.245.130.30                                                            

I can guess that the first option means "I made a mistake and need to resume testing"
I can guess that the second option means exactly what it says.
I "THINK" that the third option is the same as a disconnect in screen, leaving the tests running but disconnecting the remote connection for a bit.
What does the fourth option do?  COmpletely kill "checkbox-cli service" on the remote?
And what does the fifth do? What exactly is "Abandon the session"? Is that the same as "Kill the test run currently in progress"? Or is that the same as disconnect and forget it ever happened?

Basically, which of those is actually "Kill the test run that I just started"? And that could be messaged a bit better.

Additionally, I believe we got away from the Master/Slave terminology in favor of Remote/Service, so there's a bit of a hanger-on there.

Note, this is checkbox-snappy running on the R340, I haven't looked at current checkbox via Debs to see if this affects that too.

## Attachments
No attachments

Tags:
[]

**Comment by beliaev-maksim** [comment]
This thread was migrated from launchpad.net
##### https://launchpad.net/~pieq wrote on 2022-03-15 03:46:14:
Same issue here in the QA team. Sometimes we would like to stop the current session, but it's not clear what option we should use.

For instance, let's say we are running a stress test. After 9 failed reboots, we realize that the test run is going nowhere and there is no point running yet another 91 reboots that are going to fail. We need to stop the current session on the slave, but we still need access to the session data (io-logs, etc.), so we don't want to delete it. Like Jeff, we are not sure what the last option does exactly.

As for the 4th option, it does stop the service, but then we don't know how to resume the service afterwards... When I had the issue, I ended up restarting the device to force restarting the service... I guess there is existing doc about this[1], albeit `checkbox-snap slave restart` would be a nice command to have :)

[1] https://checkbox.readthedocs.io/en/latest/slave-daemon.html

**Comment by Hook25** [comment]
This issue was fixed in #540
